### DISCUSSION TITLE
Experimental feature: Export Puffin Reports

### DISCUSSION THREAD

**Discussion by Kerollmops** [meilisearch/product#693] [Experimental features]
<table>
    <thead>
        <tr>
            <th>Feature name</th>
  			<th>How to enable</th>
  			<th>Description</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>export Puffin reports</td>
            <td>

```
curl \
  -X PATCH 'http://localhost:7700/experimental-features/' \
  -H 'Content-Type: application/json'  \
--data-binary '{
    "exportPuffinReports": true
  }'
```
  
 </td>
  			<td>Enables exporting indexing reports</td>
        </tr>
    </tbody>
</table>

Meilisearch v1.5 (released November 20) introduces puffin report exporting.

## Experimental feature abstract

Enabling the `"exportPuffinReport": true` experimental feature will make Meilisearch automatically export `.puffin` file reports. Analyzing those files can give great information on what's happening inside the engine and how the Meilisearch team can improve the indexing performances.

The reports are exported at every batch of operation and contain a lot of colorful bubbles to describe the time spent into the indexing part of the engine. Below is an example [of Puffin Viewer](https://github.com/EmbarkStudios/puffin/tree/main#ui) displaying a Meilisearch indexing operation report.

<img width="1752" alt="profiling-example" src="https://github.com/meilisearch/product/assets/3610253/77d1cf2a-3710-45dc-a9cb-2b197b9793f2">

### How to use the feature?

1. Call the `PATCH /experimental-features` passing `"exportPuffinReports": true`.
2. Send documents, delete documents or anything that is part of the indexing process of the engine.
3. The engine should have generated `.puffin` reports in the `$CWD`. Most of the time, next to the `data.ms` file.

🗣️  **You are welcome to give feedback about the indexing performance and to provide the associated `.puffin` reports; we are eager to collect feedback on the feature**

**Comment by Fronix** [comment]
Doesn't seem to work with Windows
```
[2023-11-16T09:12:22Z ERROR index_scheduler] The filename, directory name, or volume label syntax is incorrect. (os error 123)
```

**Comment by curquiza** [comment]
This feature is not available anymore, we removed it. Replaced by logs routes and log mode
